doc: update System Errors documentation (#24090)
nodejs/node: doc/api/errors.md

## System Errors

Node.js generates system errors when exceptions occur within its runtime
environment. These usually occur when an application violates an operating

> system constraint.  For example, a system error will occur if an application

Reviewer: ```suggestion
system constraint. For example, a system error will occur if an application
```

attempts to read a file that does not exist.

System errors are usually generated at the syscall level. For a comprehensive
list, see the [`errno`(3) man page][].

In Node.js, system errors are `Error` objects with extra properties.

### Class: SystemError

> * `address` {string} If present, the address to which a networkconnection failed

Reviewer: ```suggestion
* `address` {string} If present, the address to which a network connection failed
```

* `code` {string} The string error code
* `dest` {Buffer} If present, the file path destination when reporting a file
  system error
* `errno` {number} The system-provided error number
* `info` {Object} If present, extra details about the error condition
* `message` {string} A system-provided human-readable description of the error

> * `path` {Buffer} If present, the file path When reporting a file system error

Reviewer: ```suggestion
* `path` {Buffer} If present, the file path when reporting a file system error
```

* `port` {number} If present, the network connection port that is not available
* `syscall` {string} The name of the system call that triggered the error

#### error.address

* {string}

If present, `error.address` is a string describing the address to which a

> connection failed.

Reviewer: ```suggestion
network connection failed.
```

## System Errors

Node.js generates system errors when exceptions occur within its runtime
environment. These usually occur when an application violates an operating
system constraint. For example, a system error will occur if an application
attempts to read a file that does not exist.

System errors are usually generated at the syscall level. For a comprehensive
list, see the [`errno`(3) man page][].

In Node.js, system errors are `Error` objects with extra properties.

### Class: SystemError

* `address` {string} If present, the address to which a network connection
  failed
* `code` {string} The string error code

> * `dest` {Buffer} If present, the file path destination when reporting a file

Author: Yes, to this and your other comments. Applying the suggestions now.

  system error
* `errno` {number} The system-provided error number
* `info` {Object} If present, extra details about the error condition
* `message` {string} A system-provided human-readable description of the error
* `path` {Buffer} If present, the file path when reporting a file system error
* `port` {number} If present, the network connection port that is not available
* `syscall` {string} The name of the system call that triggered the error

#### error.address

* {string}

If present, `error.address` is a string describing the address to which a
network connection failed.

#### error.code

* {string}

> The `error.code` property is a string representing the error code. `error.code`

Reviewer 2: Would be even nicer if this is linked at the top of the document instead of hiding in at `####`, but that can be done in another PR (but then this would not be properly fixing https://github.com/nodejs/node/issues/23975).

Author: I moved the note about code stability to the one under the class Error. Hopefully that fixes the issue.

system error.

#### error.errno

* {string|number}

The `error.errno` property is a number or a string. If it is a number, it is a negative value which corresponds to the error code defined in
[`libuv Error handling`]. See `uv-errno.h` header file
(`deps/uv/include/uv-errno.h` in the Node.js source tree) for details. In case
of a string, it is the same as `error.code`.

#### error.info

* {Object}

If present, `error.info` is an object with details about the error condition.

#### error.message

* {string} 

`error.message` is a system-provided human-readable description of the error.
It is subject to change between versions of Node.js. Use `code` for a more
stable indicator of the type of error.

#### error.path

* {string}

If present, `error.path` is a string containing a relevant invalid pathname.

#### error.port

* {number}

If present, `error.port` is the network connection port that is not available.

#### error.syscall

* {string}

The `error.syscall` property is a string describing the [syscall][] that failed.

### Common System Errors

This is a list of system errors commonly-encountered when writing a Node.js

> program. For a comprehensive list, see the [`errno`(3) man page][].

Reviewer 2: Now come to think of it, this seems to be a bit misleading since this list only includes the native system error codes, not our home-grown error codes - so, one may simply skip to the man page without scrolling down to see the `ERR_` list..but, it was originally written that way so ¯\\_(ツ)_/¯

Author: Moved the stability note to the general class. Hope that helps.
